src/coreclr/jit/disasm.cpp

```diff
@@ -1544,6 +1562,175 @@ void DisAssembler::disOpenForLateDisAsm(const char* curMethodName, const char* c
     disCurClassName  = curClassName;
 }
 
+#ifdef USE_COREDISTOOLS
+
+// Helper functions to print messages from CoreDisTools Library
+// The file/linenumber information is from this helper itself,
+// since we are only linking with the CoreDisTools library.
+
+// Currently all loggers are the same
+#define LOGGER(L)                                                                                                      \
+    static void __cdecl CorDisToolsLog##L(const char* msg, ...)                                                        \
+    {                                                                                                                  \
+        va_list argList;                                                                                               \
+        va_start(argList, msg);                                                                                        \
+        vflogf(disAsmFileCorDisTools, msg, argList);                                                                   \
+        va_end(argList);                                                                                               \
+        fprintf(disAsmFileCorDisTools, "\n");                                                                          \
+    }
+
+LOGGER(VERBOSE)
+LOGGER(ERROR)
+LOGGER(WARNING)
+
+const PrintControl CorPrinter = {CorDisToolsLogERROR, CorDisToolsLogWARNING, CorDisToolsLogVERBOSE,
+                                 CorDisToolsLogVERBOSE};
+
+NewDisasm_t*       g_PtrNewDisasm       = nullptr;
+DumpInstruction_t* g_PtrDumpInstruction = nullptr;
+FinishDisasm_t*    g_PtrFinishDisasm    = nullptr;
+
+// Coredistools disassembler initialization.
+//
+// Returns true on success, false on failure.
+//
+bool DisAssembler::InitCoredistoolsDisasm()
+{
+    // TODO: call g_PtrFinishDisasm sometime?
```

Comment:
still relevant?

Reply:
Yes. I went ahead and wrote some better initialization/un-initialization to fix this. Also fixed it to only load coredistools.dll once per process: it was loading it once for every function being disassembled.